rs.h

```diff
@@ -3,7 +3,9 @@
 
 #include <stdint.h>
 
+#ifndef DATA_SHARDS_MAX
 #define DATA_SHARDS_MAX 255
+#endif
```

Comment:
I would suggest leaving this as is as it's intention is to inform the maximum shards that can be used in `GF2^8`, to further reduce the size of the permutation vectors in `reed_solomon_decode()` maybe a new define needs to introduced or maybe use vla's there, though that's problematic for embedded targets as well
```suggestion
#define DATA_SHARDS_MAX 255
```

Reply:
Ok but I would prefer the VLA version then.
Shard number is not so variable and the code is only called once per shard ensemble, so the VLA cost is negligible (but I am too not a fan).

Another approach would be allocating the array in `reed_solomon_new()` but that would inflict unnececary cost on the encoder.

Reply:
> Another approach would be allocating the array in `reed_solomon_new()` but that would inflict unnececary cost on the encode-only path.

Agree, that was the reasoning behind `_decode` being the way it is. Let's go with VLA.